### PR TITLE
Fix update of coefficients in VSIMEXMethod

### PR DESCRIPTION
**Pull request by sebglane**

### CODE REVIEW

File: tests/rotatingMHD/time_discretization.output [hunk 1]
```diff
@@ -21,6 +21,69 @@ Step =      0, Current time = 0.000000e+00, Next time step = 1.000000e-01
...
+Step =     10, Current time = 1.00e+00, Next time step = 0.00e+00
+Step =     10, Current time = 1.00e+00, Next time step = 2.00e-01
+    +-------------+------------+------------+------------+
+    |    Index    |     n      |    n-1     |     n-2    |
+    +-------------+------------+------------+------------+
+    |    alpha    |   1.67e+00 |  -3.00e+00 |   1.33e+00 | 
+    |    beta     |     -      |   3.00e+00 |  -2.00e+00 | 
+    |    gamma    |   1.00e+00 |   0.00e+00 |   0.00e+00 | 
+    |  extra_pol  |     -      |   3.00e+00 |  -2.00e+00 | 
+    +-------------+------------+------------+------------+
```
Comment (j507) [inline tests/rotatingMHD/time_discretization.output:50]: The changed values are due to resuming with a new time step, correct?

File: tests/rotatingMHD/vector_tools_02.mpirun=2.output [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+H1 error: 7.50576e-15
+Linfty error: 1.55431e-15
+Vector difference: 1.48952e-15
+L2 error: 1.57316e-16
+H1 error: 2.19841e-15
+Linfty error: 4.44089e-16
+Vector difference: 1.167e-14
+L2 error: 5.03016e-16
+H1 error: 4.33345e-15
+Linfty error: 1.55431e-15
```
Comment (j507) [inline tests/rotatingMHD/vector_tools_02.mpirun=2.output:16]: Won't this output depend on the machine precision of the computer where the test is to be performed?

**Review by j507 — APPROVED** [review]

File: tests/rotatingMHD/time_discretization.output (hunk 1 above)
Comment (sebglane) [reply tests/rotatingMHD/time_discretization.output:50]: I will check this.

File: tests/rotatingMHD/vector_tools_02.mpirun=2.output (hunk 2 above)
Comment (sebglane) [reply tests/rotatingMHD/vector_tools_02.mpirun=2.output]: Yes, I will add `(error < 1e-12)` instead.

File: tests/rotatingMHD/time_discretization.output (hunk 1 above)
Comment (sebglane) [reply tests/rotatingMHD/time_discretization.output:50]: I will check this...

**Pull request merged**